bot/cogs/image_dots_cog.py

```diff
@@ -0,0 +1,199 @@
+            embed.add_field(name='Exception:', value='threshold not a valid int or in range.')
+            await ctx.send(embed=embed)
+            return
+        # intentionally not converting to a 2d array here. If someone, or myself, wants to
+        # add a 1d array functionality, great! I am sticking with 2d arrays right now. 
+        # check if we need to open a file, or a url
+        # https://github.com/FranciscoMoretti/asciify-color/blob/master/asciify.py
+        # https://stackoverflow.com/questions/7391945/how-do-i-read-image-data-from-a-url-in-python
+        new_img = []
+        try:
+            # meant when the person wants to exclude the embed in a regular message
+            # breaks the bot otherwise
+            filename = filename.replace('<', '').replace('>', '')
+            with Image.open(requests.get(filename, stream=True).raw) as img:
+                # default asciiWidth I found online. Aparently over 500 gets laggy
+                new_img = self.parse_image(img, width)
+                await ctx.send('\n'.join(new_img))
+        except UnidentifiedImageError as e:
+            embed = discord.Embed(title=f'ERROR: unable to open message link', color=Colors.Error)
+            embed.add_field(name='Exception:', value=filename + ' link does not exist, or is broken!')
+            await ctx.send(embed=embed)
+
+    @todots.command()
+    @ext.long_help('Takes any image, and returns the brailled image.\nCan specify mobile or pc width characteristics.\nDefault width is pc size.\
+                    Choose a width of \'pc\', \'mobile\', or leave blank.\n \
+                    Threshold determines which pixels are white, and which are blank. Choose a value [0-255]\n\
+                    The final argument asks, do you want to invert the image or not?\n \
+                    When attachment specifier is used, you can upload an image directly. Must attach an image.')
+    @ext.short_help('Turn an image to a braille image.\nDefault width is pc size.\nDefault threshold is 150.\n\
+                    To specify threshold you must include all required arguments.\n \
+                    The same goes for all arguments. Must attach an image')
+    @ext.example(('todots attachment [mobile|pc] [threshold = 0-255] [inverted = 0/1]',
+    'todots attachment [mobile|pc] [threshold = 0-255]',
+    'todots attachment [mobile|pc]',
+    'todots attachment'))
+    async def attachment(self, ctx, device = None, threshold = 150, inverted = 0) -> None:
+        image = ''
+        try:
+            image = ctx.message.attachments[0].url
+        except Exception as e:
+            embed = discord.Embed(title=f'ERROR: must include an attached image', color=Colors.Error)
+            embed.add_field(name='Exception:', value="upload an image when using 'attachment' specifier")
+            await ctx.send(embed=embed)
+            return
+        return await self.todots(ctx, image, device, threshold, inverted)
```

Comment:
have both commands call a todots function, a command shouldnt invoke a different command

Reply:
so drag the common functionality to both commands into its own functions